Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -105,9 +105,36 @@ public override string DataSource
 
         public override ConnectionState State => _connectionState;
         internal SnowflakeDbTransaction ExplicitTransaction { get; set; } // tracks only explicit transaction operations
+
+        public void PreventFromReturningToPool()
+        {
+            if (SfSession == null)
+            {
+                throw new Exception("Connection is not ready to prevent its session from returning to the pool");

```

Comment:
It would be more descriptive, however revealing less details has also some pros.
But I changed to: `Session not yet created for this connection. Unable to prevent the session from pooling`